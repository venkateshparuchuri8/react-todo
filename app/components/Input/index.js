import styled, { css } from 'styled-components';

const Input = styled.input`
  margin: 0;
  max-width: 100%;
  -webkit-box-flex: 1;
  -ms-flex: 1 0 auto;
  flex: 1 0 auto;
  outline: 0;
  -webkit-tap-highlight-color: rgba(255,255,255,0);
  text-align: left;
  line-height: 1.21428571em;
  padding: .67857143em 1em;
  background: transparent;
  border:none;
  border-bottom: 1px solid rgba(34,36,38,.15);
  color: rgba(0,0,0,.87);
  -webkit-transition: box-shadow .1s ease,border-color .1s ease;
  transition: box-shadow .1s ease,border-color .1s ease;
  box-shadow: none;
  & :focus {
    border-color: #85b7d9;
    background: #fff;
    color: rgba(0,0,0,.8);
    box-shadow: none;
  }
  ${({ fluid }) => fluid && css`
    width: auto;
  `}
  ${({ border }) => {
    if (border === 'border') {
      return css`
        border: 1px solid rgba(34,36,38,.15) !important;
      `;
    }
    if (border === 'noBorder') {
      return css`
        background-color: ${(props) => props.theme.secondaryColor};
        color: #fff;
      `;
    }
    return '';
  }}
`;
Input.displayName = 'input';
export default Input;
