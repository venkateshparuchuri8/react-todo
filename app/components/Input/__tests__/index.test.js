/**
 * @author vishwadeep.kapoor
 * @Date: 2018-12-26 17:41:54
 * @Last Modified by: vishwadeep
 * @Last Modified time: 2018-12-26 17:44:31
 */
/* eslint-disable no-undef */
import React from 'react';
import Input from '../index';

describe('Input rendering', () => {
  it('is renders correctly with border', () => {
    const props = { border: 'border' };
    const tree = renderWithTheme(<Input {...props} />);
    expect(tree).toMatchSnapshot();
  });
  it('is renders correctly with no-border', () => {
    const props = { border: 'noBorder' };
    const tree = renderWithTheme(<Input {...props} />);
    expect(tree).toMatchSnapshot();
  });
});
