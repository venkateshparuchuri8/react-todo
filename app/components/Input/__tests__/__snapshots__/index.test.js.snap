// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input rendering is renders correctly with border 1`] = `
<input
  className="sc-bdVaJa lWlPK"
/>
`;

exports[`Input rendering is renders correctly with no-border 1`] = `
<input
  className="sc-bdVaJa EKMUm"
/>
`;
